src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerConfig.java

```diff
@@ -29,8 +32,11 @@
   public static final String AUDITOR_CLASS_CONFIG = "auditor.class";
   public static final String KEY_SERIALIZER_CLASS_CONFIG = ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
   public static final String VALUE_SERIALIZER_CLASS_CONFIG = ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
-  public static final String SEGMENT_SERIALIZER_CLASS_CONFIG = "segment.serializer";
   public static final String CURRENT_PRODUCER = "current.producer";
+  public static final String UUID_FACTORY_CLASS_CONFIG = "uuid.factory.class";
+  public static final String HEADER_PARSER_CONFIG = LiKafkaConsumerConfig.HEADER_PARSER_CONFIG;
+
+  public static final String HEADER_PARSER_DOC = LiKafkaConsumerConfig.HEADER_PARSER_DOC;
```

Comment:
There is probably not substantial difference between the two. But there might be some slight benefits to separate the interface.

First of all, having separate interface does not prevent people from putting the serde code in the same class if they want to. i.e. implement both interface.

The benefit of having separate interface is that users are not forced to implement both serializer and deserializer if they only want to have one of them, but not both. Imagine we are doing migration and currently we have old wire protocol and now we are going to switch to the new wire protocol with new producer. Likely we only need to let the consumer to deserialize with both new and old wire protocol, so only a customized deserializer is needed, but if we have only one serde interface, users will be forced to implement the serializer as well, which is not ideal.

The configuration on producer and consumer are likely in different processes, so they will have to be configured separately even if the serde code is in the same class. So having serde code in the same class does not really avoid misconfiguration. It is probably more important to have corresponding tests to make sure everything works fine end to end instead of relying on the colocation of the code.